### PR TITLE
UnitTest Coverage for ARP

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
UnitTest Coverage for ARP

Description
-----------
This PR adds Unit Test for ARP.c , coverage 99.2% due to source file issue, described in the vARPProcessPacketRequest API

Test Steps
-----------
```
cmake -S test/unit-test -B test/unit-test/build/ -DCMAKE_BUILD_TYPE=Debug
make -C test/unit-test/build/ all
```

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ActoryOu** [comment]
The coverage is not 100%. Could you help to check?
`FreeRTOS_ARP.c                            |98.7%    390| 100%    20|99.2%    240`

File: source/FreeRTOS_ARP.c [hunk 1]
```diff
@@ -290,7 +290,7 @@ eFrameProcessingResult_t eARPProcessPacket( const NetworkBufferDescriptor_t * px
                 {
                     case ipARP_REQUEST:
 
-                        if( ( ulTargetProtocolAddress == pxTargetEndPoint->ipv4_settings.ulIPAddress ) &&
+                        if( ( ulTargetProtocolAddress == pxTargetEndPoint->ipv4_settings.ulIPAddress ) && /* This check seems unnecessary which makes the first if condition in Line 356 obsolete. */
```
Comment (ActoryOu) [inline source/FreeRTOS_ARP.c:293]: I don't get it, it seems like IP address is possible to be different.

File: source/FreeRTOS_ARP.c [hunk 2]
```diff
@@ -925,7 +925,7 @@ static BaseType_t prvFindCacheEntry( const MACAddress_t * pxMACAddress,
                  pxEndPoint != NULL;
                  pxEndPoint = FreeRTOS_NextEndPoint( NULL, pxEndPoint ) )
             {
-                if( ENDPOINT_IS_IPv4( pxEndPoint ) )
+                if( ( pxEndPoint )->bits.bIPv6 == 0U ) /*NULL End Point is checked in the for loop, no need for an extra check */
```
Comment (ActoryOu) [inline source/FreeRTOS_ARP.c:928]:
```suggestion
                if( pxEndPoint->bits.bIPv6 == 0U ) /*NULL End Point is checked in the for loop, no need for an extra check */
```
Please remove redundant bracket.

**Review by ActoryOu — CHANGES_REQUESTED** [review]

File: source/FreeRTOS_ARP.c (hunk 2 above)
Comment (kar-rahul-aws) [reply source/FreeRTOS_ARP.c]: updated

File: source/FreeRTOS_ARP.c (hunk 1 above)
Comment (kar-rahul-aws) [reply source/FreeRTOS_ARP.c:293]: updated

Comment (ActoryOu) [reply source/FreeRTOS_ARP.c:293]: We probably can remove the comment here.

**Review by ActoryOu — APPROVED** [review]

**Review by ActoryOu — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Review by tony-josi-aws — APPROVED** [review]

**Pull request merged**